### PR TITLE
Bucket encryption checkbox and endpoints for Console

### PR DESCRIPTION
**Pull request by Alevsk**
Screen Shot 2020-10-23 at 4 46 11 PM
Screen Shot 2020-10-23 at 4 46 20 PM
Screen Shot 2020-10-23 at 4 46 02 PM
Screen Shot 2020-10-23 at 4 46 06 PM

### CODE REVIEW

File: restapi/user_buckets.go [hunk 1]
```diff
@@ -462,3 +485,84 @@ func consoleAccess2policyAccess(bucketAccess models.BucketAccess) (bucketPolicy
...
 }
+
+// enableBucketEncryption will enable bucket encryption based on two encryption algorithms, sse-s3 (server side encryption with external KMS) or sse-kms (aws s3 kms key)
+func enableBucketEncryption(ctx context.Context, client MinioClient, bucketName string, encryptionType models.BucketEncryptionType, kmsKeyID string) error {
+	var config *sse.Configuration
+	switch strings.ToLower(string(encryptionType)) {
+	case "sse-kms":
+		config = sse.NewConfigurationSSEKMS(kmsKeyID)
+	case "sse-s3":
+		config = sse.NewConfigurationSSES3()
```
Comment (cesnietor) [inline restapi/user_buckets.go:496]: here you can use the already defined types:
```
const (

	// BucketEncryptionTypeSseS3 captures enum value "sse-s3"
	BucketEncryptionTypeSseS3 BucketEncryptionType = "sse-s3"

	// BucketEncryptionTypeSseKms captures enum value "sse-kms"
	BucketEncryptionTypeSseKms BucketEncryptionType = "sse-kms"
)

```

File: swagger.yml [hunk 2]
```diff
@@ -1779,6 +1850,22 @@ definitions:
       creation_date:
         type: string
 
+  bucketEncryptionRequest:
+    type: object
+    properties:
+      encType:
+        $ref: "#/definitions/bucketEncryptionType"
+      kmsKeyID:
+        type: string
```
Comment (cesnietor) [inline swagger.yml:1859]: are this optional or required? I think they must be required.

Comment (Alevsk) [reply swagger.yml:1859]: They are optional, default value for enc type will be sse-s3

**Review by cesnietor — APPROVED** [review]
LGTM

**Review by dvaldivia — APPROVED** [review]

**Pull request merged**